File: backend/app/schemas/itinerary.py
from pydantic import BaseModel


class ItineraryRequest(BaseModel):
    destination: str
    budget: float  # In INR (₹)
    days: int
    interests: list[str]


class DayPlan(BaseModel):
    day: int
    title: str
    activities: list[str]
    estimated_cost_inr: float
    estimated_cost_local: float = 0.0


class BudgetBreakdown(BaseModel):
    accommodation: float
    food: float
    transport: float
    activities: float
    miscellaneous: float
    total: float


class ItineraryResponse(BaseModel):
    destination: str
    destination_country: str = "Destination Country"
    currency_code: str = "INR"
    currency_symbol: str = "₹"
    exchange_rate_info: str = "1 INR = 1 INR"
    exchange_rate_to_inr: float = 1.0
    days: int
    itinerary: list[DayPlan]
    budget_breakdown: BudgetBreakdown
    packing_list: list[str]
    travel_tips: list[str]

File: src/api/itinerary.ts
import client from "./client";

export interface DayPlan {
  day: number;
  title: string;
  activities: string[];
  estimated_cost_inr: number;
  estimated_cost_local?: number;
}

export interface BudgetBreakdown {
  accommodation: number;
  food: number;
  transport: number;
  activities: number;
  miscellaneous: number;
  total: number;
}

export interface ItineraryResponse {
  destination: string;
  destination_country?: string;
  currency_code?: string;
  currency_symbol?: string;
  exchange_rate_info?: string;
  exchange_rate_to_inr?: number;
  days: number;
  itinerary: DayPlan[];
  budget_breakdown: BudgetBreakdown;
  packing_list: string[];
  travel_tips: string[];
}


export interface ItineraryRequest {
  destination: string;
  budget: number;
  days: number;
  interests: string[];
}

export const itineraryApi = {
  generate: (data: ItineraryRequest) =>
    client.post<ItineraryResponse>("/trips/generate-itinerary", data),
  exportPdf: (data: ItineraryResponse) =>
    client.post("/trips/export-pdf", data, { responseType: "blob" }),
};

